packages/router-generator/src/generator.ts

```diff
@@ -380,20 +380,27 @@ export async function generator(config: Config) {
         path.relative(
           path.dirname(config.generatedRouteTree),
           path.resolve(config.routesDirectory, routePathIdPrefix + rootPathId),
-        ),
+        ) + config.addExtensions
+          ? '.tsx'
+          : '',
       )}'`,
       ...sortedRouteNodes
         .filter((d) => !d.isVirtual)
         .map((node) => {
           return `import { Route as ${
             node.variableName
           }Import } from './${replaceBackslash(
-            removeExt(
-              path.relative(
-                path.dirname(config.generatedRouteTree),
-                path.resolve(config.routesDirectory, node.filePath),
-              ),
-            ),
+            config.addExtensions
```

Comment:
to avoid code duplication, I would modify a function like this:

```tsx
export function removeExt(d: string, enabled:boolean = true) {
  return enabled ? d.substring(0, d.lastIndexOf('.')) || d : d
}
```
and then call removeExt with `config.addExtensions`

you could also create a new wrapping function around `removeExt` to separate those concerns

Reply:
Thanks for pointing this, i pushed another version.